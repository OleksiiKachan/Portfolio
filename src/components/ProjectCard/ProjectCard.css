.projectCard {
    width: 45%;
    min-width: 300px;
    min-height: 320px;
    margin: 5px;
	padding: 0;
    color: white;
    box-sizing: border-box;
    perspective: 1000px;
}

.projectCard__description {
    font-weight: lighter;
    font-size: smaller;
}

.projectCard__innerArea {
    min-height: 320px;
    position: relative;
    width: 100%;
    height: 100%;
    text-align: center;
    transition: transform 0.5s;
    transform-style: preserve-3d;
    box-shadow: 2px 2px 20px 0px rgba(0,0,0,0.29);
}

.projectCard:hover .projectCard__innerArea {
    transform: rotateY(180deg);
    box-shadow: 0 0 0 0 rgba(0,0,0,0.29);
}

.projectCard__front, .projectCard__back {
    position: absolute;
    width: 100%;
    height: 100%;
    backface-visibility: hidden;
}

.projectCard__front {
	background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
}

.projectCard__shader {
    background-color: black;
    opacity: 0.3;
	position: relative;
    height: 100%;
}

.projectCard_hyperloop {
    background-image: url('../../images/projects/hyperloop.png');
}

.projectCard_branding_oleksiiKachan {
    background-image: url('../../images/projects/branding_oleksiiKachan.png');
}

.projectCard_branding_irynaKachan {
    background-image: url('../../images/projects/branding_irynaKachan.png');
}

.projectCard_branding_oleksandrGorchakov {
    background-image: url('../../images/projects/branding_oleksandrGorchakov.png');
}

.projectCard_ui_networkControl {
    background-image: url('../../images/projects/ui_networkControl.png');
}

.projectCard_corrugatedCardboard {
    background-image: url('../../images/projects/corrugatedCardboard.png');
}

.projectCard_ftpService {
    background-image: url('../../images/projects/ftpService.png');
}

.projectCard_agw {
    background-image: url('../../images/projects/agw.png');
}

.projectCard_tdm {
    background-image: url('../../images/projects/tdm.png');
}


.projectCard__back {
    transform: rotateY(180deg);
    padding: 20px;
    height: calc(100% - 40px);
    width: calc(100% - 40px);
	display: flex;
	flex-direction: column;
    justify-content: space-between;
}

.projectCard__solidBackground_red {
    background-color: #7e0000;
}

.projectCard__solidBackground_blue {
    background-color: #00187e;
}

.projectCard__solidBackground_grey {
    background-color: #474747;
}

.projectCard__solidBackground_green {
    background-color: #004a02;
}

.projectCard__solidBackground_yellow {
    background-color: #514704;
}

.projectCard__title {
    margin-bottom: 10px;
}

.projectCard__description {
    font-weight: lighter;
    margin-bottom: 10px;
}

.toolsList {
    list-style-type: none;
    display: flex;
}

.toolsList__toolIcon {
    margin-right: 5px;
}

.projectCard__downloadFile {
    text-align: left;
}

.projectCard__downloadLink {
    text-decoration: none;
    color: #647ac3;
}

.projectCard__downloadLink:hover {
    text-decoration: underline;
}

.projectCard__downloadIcon {
    margin-left: 15px;
}

.projectCard__downloadIcon_hidden {
    display: none;
}