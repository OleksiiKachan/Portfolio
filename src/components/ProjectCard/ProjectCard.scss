.projectCard {
  width: 45%;
  min-width: 300px;
  min-height: 320px;
  margin: 5px;
  padding: 0;
  color: white;
  box-sizing: border-box;
  perspective: 1000px;
}

.projectCard__innerArea {
  min-height: 320px;
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: transform 0.5s;
  transform-style: preserve-3d;
  box-shadow: 2px 2px 20px 0px rgba(0, 0, 0, 0.29);
}

.projectCard:hover .projectCard__innerArea {
  transform: rotateY(180deg);
  box-shadow: 0 0 0 0 rgba(0, 0, 0, 0.29);
}

.projectCard:hover .projectCard__shader {
  transition: visibility 0.5s;
  visibility: collapse;
}

.projectCard__front {
  position: absolute;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
}

.projectCard__front {
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
}

.projectCard__shader {
  background-color: black;
  opacity: 0.3;
  position: relative;
  height: 100%;
}
