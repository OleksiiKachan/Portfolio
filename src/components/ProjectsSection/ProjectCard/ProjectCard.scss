@import '../../../assets/style/variables';

.projectCard {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  margin: 3rem;
  width: 25%;
  overflow: hidden;
  background-size: cover;

  &__inner {
    opacity: 0;
    background-color: $color-black;
    display: flex;
    flex-direction: column;
    align-items: center;

    &:hover {
      animation: opacity 200ms ease-in-out forwards;

      .projectCard__title {
        animation: fromLeftToCenter 500ms ease-in-out forwards;
      }
      .projectCard__stack {
        animation: fromRightToCenter 500ms ease-in-out forwards;
      }
      .projectCard__button {
        animation: fromBottomToCenter 500ms ease-in-out forwards;
      }
    }
  }

  &__title {
    display: flex;
    align-self: flex-start;
    max-width: 85%;
    background: rgba(221, 221, 221, 0.9);
    border-radius: 0px 0px 50px 0px;
    padding: 0.5rem 1.85rem 0.5rem 0.5rem;
    font-size: 0.85rem;
    font-weight: 300;
    color: $color-black;

    transform: translateX(-100%);
  }

  &__stack {
    color: $color-white;
    font-weight: 500;
    padding: 20%;
    text-align: center;

    transform: translateX(-100%);
  }

  &__button {
    margin-bottom: 2rem;

    transform: translateY(100%);
  }
}

@media screen and (max-width: 825px) {
  .projectCard {
    margin: 1rem;
    width: 80%;

    &__inner {
      opacity: 1;
      background-color: rgba($color-black, 0.85);

      &:hover {
        animation: none;

        .projectCard__title {
          animation: none;
        }
        .projectCard__stack {
          animation: none;
        }
        .projectCard__button {
          animation: none;
        }
      }
    }

    &__title {
      transform: translateX(0);
    }

    &__stack {
      transform: translateX(0);
    }

    &__button {
      transform: translateY(0);
    }
  }
}
