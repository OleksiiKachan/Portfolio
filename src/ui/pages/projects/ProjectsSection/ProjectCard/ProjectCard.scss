@import '../../../../assets/style/variables';
@import '../../../../assets/style/mixins';

.projectCard {
  overflow: hidden;
  display: flex;

  &__coverImage {
    height: 100%;
    width: 60%;
  }

  &__projectInfo {
    width: 40%;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 1rem;
    text-align: center;
  }

  &__title {
    padding: 1rem 0;
    letter-spacing: 0.05ch;

    @include respond(tablet) {
      padding: 0;
      font-size: 0.9rem;
    }
  }

  &__stackList {
    width: 100%;
    display: flex;
    justify-content: space-around;
    padding-bottom: 1rem;

    @include respond(tablet) {
      display: none;
    }
  }

  &__description {
    font-size: 1rem;
    letter-spacing: 0.01ch;
    font-weight: 400;
    padding: 1.8rem 0;
  }

  &__button {
    padding-top: 3rem;
    color: white;
  }
}
