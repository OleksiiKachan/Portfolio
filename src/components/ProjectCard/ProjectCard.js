import React, { Component } from 'react';
import classNames from 'classnames';
import './ProjectCard.css';
import ProjectDescription from '../ProjectDescription';

/*
    <ProjectCard project={} className=''/>
*/
class ProjectCard extends Component {
    render() {
        return (
            <div className={classNames('projectCard', this.props.className)}>
                <div className={classNames('projectCard__innerArea')}>
                    <div className={classNames('projectCard__front', `projectCard_${this.props.project.fileName}`)}>
                        <div className={classNames('projectCard__shader')}>
                        </div>
                    </div>
                    <ProjectDescription project={this.props.project} className=''/>
                </div>
            </div>
        );
    }
}

export default ProjectCard;