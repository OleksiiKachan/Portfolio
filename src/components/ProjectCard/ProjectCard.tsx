import React, { Component } from 'react';
import classNames from 'classnames';
import './ProjectCard.scss';

/*
    <ProjectCard project={} className=''/>
*/
class ProjectCard extends Component<any, any> {
  render() {
    var cardBackground = {
      backgroundImage: `url(https://res.cloudinary.com/oleksiikachan/image/upload/v1543467854/portfolio/images/projects/${
        this.props.project.fileName
      }.png)`,
    };

    return (
      <div
        className={classNames('projectCard', this.props.className)}
        style={cardBackground}
      >
        <div className={classNames('projectCard__content')}>
          <div className={classNames('projectCard__titleSection')}>
            <h3 className={classNames('projectCard__title')}>
              {this.props.project.title}
            </h3>
          </div>
          <div className={classNames('projectCard__technologiesSection')}>
            <h4 className={classNames('projectCard__technologies')}>
              {this.props.project.displayTechnologies}
            </h4>
            <button className={classNames('projectCard__button')}>
              See More
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default ProjectCard;
