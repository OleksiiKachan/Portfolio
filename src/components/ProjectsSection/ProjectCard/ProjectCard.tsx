import React from 'react';
import classNames from 'classnames';
import OutlineButton from '../../OutlineButton';
import './ProjectCard.scss';
import { Link } from 'react-router-dom';

/*
    <ProjectCard
      title=''
      stack=''
      coverImage=''
      className=''
    />
*/

type PropsType = {
  id: string;
  title: string;
  stack: string;
  coverImage: string;
  className?: string;
};

export default ({ id, title, stack, coverImage, className }: PropsType) => {
  return (
    <div
      className={classNames('projectCard', className)}
      style={{ backgroundImage: `url(${coverImage})` }}
    >
      <div className={classNames('projectCard__shader')}>
        <div className={classNames('projectCard__inner')}>
          <div className={classNames('projectCard__title')}>{title}</div>
          <div className={classNames('projectCard__stack')}>{stack}</div>
          <Link to={`projects/${id}`}>
            <OutlineButton
              className={classNames('projectCard__button')}
              type="gold"
              onClick={() => {}}
            >
              See More
            </OutlineButton>
          </Link>
        </div>
      </div>
    </div>
  );
};
