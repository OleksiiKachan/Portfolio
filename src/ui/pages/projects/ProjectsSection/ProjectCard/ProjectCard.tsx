import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { OutlineButton, TechBadge } from '../../../../elements';
import './ProjectCard.scss';

/*
    <ProjectCard
      title=''
      stack=''
      cardImage=''
      className=''
    />
*/

type PropsType = {
  id: string;
  title: string;
  stack: Array<string>;
  cardImage: string;
  description: string;
  settings: { color: string };
  style?: any;
  className?: string;
};

export default ({
  id,
  title,
  stack,
  cardImage,
  description,
  settings,
  style,
  className,
}: PropsType) => {
  return (
    <div
      style={{ ...style, backgroundColor: settings.color }}
      className={classNames('projectCard', className)}
    >
      <img
        src={cardImage}
        alt=""
        className={classNames('projectCard__coverImage')}
      />
      <div
        style={{ backgroundColor: settings.color, color: '#fff' }}
        className={classNames('projectCard__projectInfo')}
      >
        <h3 className={classNames('projectCard__title')}>{title}</h3>
        <div className={classNames('projectCard__stackList')}>
          {stack.map(item => {
            return (
              <TechBadge
                type="light"
                key={item}
                style={{
                  animationDelay: `${800 +
                    (stack.indexOf(item) / 1.5) * 100}ms`,
                }}
                className={classNames('projectCard__stackItem')}
              >
                {item}
              </TechBadge>
            );
          })}
        </div>
        <p className={classNames('projectCard__description')}>{description}</p>
        <Link
          to={`projects/${id}`}
          className={classNames('projectCard__button')}
        >
          <OutlineButton type="white" onClick={() => {}}>
            See More
          </OutlineButton>
        </Link>
      </div>
    </div>
  );
};
