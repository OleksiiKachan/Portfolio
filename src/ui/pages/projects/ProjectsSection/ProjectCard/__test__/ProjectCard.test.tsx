import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from '../ProjectCard';

describe('ProjectCard', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ProjectCard
        id="id"
        title="AGW MES / WMS for automotive industry"
        stack={['.NET Framework']}
        settings={{ color: '#fff' }}
        description="description"
        cardImage="imageUrl"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
