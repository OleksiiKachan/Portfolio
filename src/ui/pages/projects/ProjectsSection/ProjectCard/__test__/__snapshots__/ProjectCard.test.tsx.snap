// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard Snapshot 1`] = `
<div
  className="projectCard"
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <img
    alt=""
    className="projectCard__coverImage"
    src="imageUrl"
  />
  <div
    className="projectCard__projectInfo"
    style={
      Object {
        "backgroundColor": "#fff",
        "color": "#fff",
      }
    }
  >
    <h3
      className="projectCard__title"
    >
      AGW MES / WMS for automotive industry
    </h3>
    <div
      className="projectCard__stackList"
    >
      <_default
        className="projectCard__stackItem"
        type="light"
      >
        .NET Framework
      </_default>
    </div>
    <p
      className="projectCard__description"
    >
      description
    </p>
    <Link
      className="projectCard__button"
      to="projects/id"
    >
      <_default
        onClick={[Function]}
        type="white"
      >
        See More
      </_default>
    </Link>
  </div>
</div>
`;
